pkg/injector/health_probes.go

```diff
@@ -20,8 +20,9 @@ const (
 var errNoMatchingPort = errors.New("no matching port")
 
 type healthProbe struct {
-	path string
-	port int32
+	path   string
+	port   int32
+	isHTTP bool
```

Comment:
This struct needs comments. I get the code but it might not be obvious to everyone what to expect for each probe protocol.

Reply:
I added a comment for the `isHTTP` field. I think path and port are self-explanatory.